### ISSUE TITLE
serialize error

### ISSUE THREAD

**Issue by nblog** [xiofee/protodec#2]
[message tmp{};
return tmp;](https://github.com/xiofee/protodec/blob/master/proto.hpp#L584)

DEBUG mode:
[if (!_field)](https://github.com/xiofee/protodec/blob/master/proto.hpp#L470)  fill "0xCC"
lead to wrong judgment.

sample code:
```
proto::message msg;
msg.append_child(proto::message(proto::types::type_binary, 1, "hello"));
auto buffer = msg.serialize();
```
DEBUG mode:
buffer = ""

**Comment by xiofee** [comment]
Hi, which compiler you using? return a temporary value is not good , so there is a fix 3c2a7b4, you can try this.
Thank your report! 🍺

**Comment by nblog** [comment]
It's awesome. :thumbsup:
The new version is also compatible with 64bit. :satisfied:
